driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideEncryptionAwsCredentialFromEnvironmentTest.java

```diff
@@ -192,6 +196,60 @@ public void shouldThrowMongoConfigurationIfSupplierReturnsDoesSomethingUnexpecte
         }
     }
 
+    @Test
+    public void shouldThrowMongoCryptIfNamedKmsProviderHasEmptyParametersInEncryptionSettings() {
+        assumeTrue(serverVersionAtLeast(4, 2));
+        assumeFalse(System.getenv().containsKey("AWS_ACCESS_KEY_ID"));
+        assumeTrue(isClientSideEncryptionTest());
+
+        Map<String, Map<String, Object>> kmsProviders = new HashMap<String, Map<String, Object>>() {{
+            put("aws:name", new HashMap<>());
+        }};
+
+        Map<String, Supplier<Map<String, Object>>> kmsProviderPropertySuppliers = Mockito.mock(Map.class);
```

Comment:
[optional]

If we use `MongoMockito.mock` instead, the mock itself will throw as soon as it is being interacted with. This way we avoid ever having silent incorrect behavior with no clear indication of what caused it (the default behavior of mocks may not cause an immediate exception, but instead cause the code to behave weirdly, or throw from a place that is far away from interacting with a mock).

The same applies to `shouldThrowMongoCryptIfNamedKmsProviderHasEmptyParametersInAutoSettings`.